warehouse/query-core/src/main/java/datawave/query/util/AttributeComparator.java

```diff
@@ -0,0 +1,203 @@
+                            previouslyMerged.add(attr2);
+                            
+                            // keep track of whether or not we merged an Attribute for later
+                            if (combinedAttrSet.contains(attr1)) {
+                                combinedAttrSet.remove(attr1);
+                            }
+                            if (combinedAttrSet.contains(attr2)) {
+                                combinedAttrSet.remove(attr2);
+                            }
+                            
+                            containsMatch.set(true);
+                        }
+                    } else if (attrs2.getAttributes().size() > 1 && !trueDuplicates.contains(attr2)) {
+                        if (!combinedAttrSet.contains(attr2)) {
+                            attr2.setToKeep(true);
+                            combinedAttrSet.add(attr2);
+                        }
+                    }
+                });
+            if (!containsMatch.get() && attrs1.getAttributes().size() > 1 && !trueDuplicates.contains(attr1)) {
+                if (!combinedAttrSet.contains(attr1)) {
+                    combinedAttrSet.add(attr1);
+                }
+            }
+        });
+        
+        // remove any previously "unique" attribute previously added but has now been merged
+        Iterator<Attribute> previousItr = previouslyMerged.iterator();
+        while (previousItr.hasNext()) {
+            combinedAttrSet.remove(previousItr.next());
+        }
+        
+        return combinedAttrSet;
+    }
+    
+    public static Set<Attribute<? extends Comparable<?>>> combineMultipleAttributes(final Attribute attr, final Attributes attrs, final boolean trackSizes) {
+        HashSet<Attribute<? extends Comparable<?>>> attrSet = Sets.newHashSet();
+        attrSet.add(attr);
+        
+        return combineMultipleAttributes(new Attributes(attrSet, attrs.isToKeep(), trackSizes), attrs);
+    }
+    
+    public static Set<Attribute<? extends Comparable<?>>> combineMultipleAttributes(final Attributes attrs, final Attribute attr, final boolean trackSizes) {
+        HashSet<Attribute<? extends Comparable<?>>> attrSet = Sets.newHashSet();
+        attrSet.add(attr);
+        
+        return combineMultipleAttributes(attrs, new Attributes(attrSet, attrs.isToKeep(), trackSizes));
+    }
+    
+    /**
+     * Compares two individual Attribute's flags and metadata, preferring to keep the Attribute from the event rather than the index. If the origin of the
+     * Attributes being compared is identical, prefer to keep the one with the toKeep() flag set.
+     *
+     * @param attr1
+     *            An Attribute against which we want to compare
+     * @param attr2
+     *            Another Attribute against which we want to check for likeness
+     * @return
+     */
+    public static Attribute combineSingleAttributes(final Attribute attr1, final Attribute attr2) {
```

Comment:
Is this method ever called with two logically different attributes? In that case should this return an instance of Attributes that contains both of the args?

Reply:
Pathway into this method has been updated to ensure it is only ever called when we know we have distinctly different metadata - specifically differently populated column family information.